hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/replicas/ReplicasVerify.java

```diff
@@ -88,41 +110,130 @@ void findCandidateKeys(OzoneClient ozoneClient, OzoneAddress address) throws IOE
+    ArrayNode keysArray = mapper.createArrayNode();
+
     if (!keyName.isEmpty()) {
       OzoneKeyDetails keyDetails = ozoneClient.getProxy().getKeyDetails(volumeName, bucketName, keyName);
-      processKey(keyDetails);
+      processKey(ozoneClient, keyDetails, keysArray);
     } else if (!bucketName.isEmpty()) {
       OzoneVolume volume = objectStore.getVolume(volumeName);
       OzoneBucket bucket = volume.getBucket(bucketName);
-      checkBucket(bucket);
+      checkBucket(ozoneClient, bucket, keysArray);
     } else if (!volumeName.isEmpty()) {
       OzoneVolume volume = objectStore.getVolume(volumeName);
-      checkVolume(volume);
+      checkVolume(ozoneClient, volume, keysArray);
     } else {
       for (Iterator<? extends OzoneVolume> it = objectStore.listVolumes(null); it.hasNext();) {
-        checkVolume(it.next());
+        checkVolume(ozoneClient, it.next(), keysArray);
       }
     }
+
+    root.set("keys", keysArray);
+    System.out.println(mapper.writerWithDefaultPrettyPrinter().writeValueAsString(root));
   }
 
-  void checkVolume(OzoneVolume volume) throws IOException {
+  void checkVolume(OzoneClient ozoneClient, OzoneVolume volume, ArrayNode keysArray) throws IOException {
     for (Iterator<? extends OzoneBucket> it = volume.listBuckets(null); it.hasNext();) {
       OzoneBucket bucket = it.next();
-      checkBucket(bucket);
+      checkBucket(ozoneClient, bucket, keysArray);
     }
   }
 
-  void checkBucket(OzoneBucket bucket) throws IOException {
+  void checkBucket(OzoneClient ozoneClient, OzoneBucket bucket, ArrayNode keysArray) throws IOException {
     for (Iterator<? extends OzoneKey> it = bucket.listKeys(null); it.hasNext();) {
       OzoneKey key = it.next();
       // TODO: Remove this check once HDDS-12094 is fixed
       if (!key.getName().endsWith("/")) {
-        processKey(bucket.getKey(key.getName()));
+        processKey(ozoneClient, bucket.getKey(key.getName()), keysArray);
       }
     }
   }
 
-  void processKey(OzoneKeyDetails keyDetails) {
-    replicaVerifiers.forEach(verifier -> verifier.verifyKey(keyDetails));
+  void processKey(OzoneClient ozoneClient, OzoneKeyDetails keyDetails, ArrayNode keysArray) {
+    ObjectMapper mapper = new ObjectMapper();
+
+    String volumeName = keyDetails.getVolumeName();
+    String bucketName = keyDetails.getBucketName();
+    String keyName = keyDetails.getName();
+
+    ObjectNode keyNode = JsonUtils.createObjectNode(null);
+    keyNode.put("volumeName", volumeName);
+    keyNode.put("bucketName", bucketName);
+    keyNode.put("name", keyName);
+
+    ArrayNode blocksArray = mapper.createArrayNode();
+    boolean keyPass = true; //to check if all checks are passed
+
+    try {
+      ClientProtocol clientProtocol = ozoneClient.getObjectStore().getClientProxy();
+      Map<OmKeyLocationInfo, Map<DatanodeDetails, OzoneInputStream>> replicas =
+          clientProtocol.getKeysEveryReplicas(volumeName, bucketName, keyName);
```

Comment:
I think we should make [RpcClient#getKeyInfo](https://github.com/apache/ozone/blob/a0c07c5a0aae76b2249370cdd6a305e690417210/hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java#L1853) public and that should be the only call we make to the OM for each key we get from the listing.

The ability to specify whether or not to use the container cache will be helpful for HDDS-12208. This could be used to identify if there is a stale cache bug that is causing replicas to not be returned to the client.